org.adempiere.pipo.handlers/src/org/adempiere/pipo2/handler/ReportViewColumnElementHandler.java

```diff
@@ -0,0 +1,148 @@
+					logImportDetail(ctx, impDetail, 0, "" + mReportviewColumn.getAD_ReportView_ID(),
+							mReportviewColumn.get_ID(),action);
+					throw new POSaveFailedException("Failed to save ReportViewColumn");
+				}
+			}
+		} else {
+			element.skip = true;
+		}
+	}
+
+	public void endElement(PIPOContext ctx, Element element) throws SAXException {
+	}
+
+	public void create(PIPOContext ctx, TransformerHandler document)
+			throws SAXException {
+		
+		int AD_ReportView_ID = Env.getContextAsInt(ctx.ctx, X_AD_ReportView_Column.COLUMNNAME_AD_ReportView_ID);
+		int AD_Column_ID = Env.getContextAsInt(ctx.ctx, X_AD_ReportView_Column.COLUMNNAME_AD_Column_ID);
+		
+		X_AD_ReportView_Column po = null;
+		Query query = new Query(ctx.ctx, "AD_ReportView_Column", "AD_ReportView_ID=? AND AD_Column_ID=?", getTrxName(ctx));
+		po = query.setParameters(new Object[]{AD_ReportView_ID, AD_Column_ID}).first();
+		System.out.println("po:" + po.toString());
+		if (po != null) {
+
+			if (!isPackOutElement(ctx, po))
+				return;
+			
+			verifyPackOutRequirement(po);
+			
+			AttributesImpl atts = new AttributesImpl();
+			addTypeName(atts, "table");
+			document.startElement("", "", X_AD_ReportView_Column.Table_Name, atts);
+			createReportViewColumnBinding(ctx, document, po);
+			document.endElement("", "", X_AD_ReportView_Column.Table_Name);
+		}
+	}
+
+	private void createReportViewColumnBinding(PIPOContext ctx, TransformerHandler document,
+			X_AD_ReportView_Column m_Reportview_Column) {
+
+		PoExporter filler = new PoExporter(ctx, document, m_Reportview_Column);
+		List<String> excludes = defaultExcludeList(X_AD_ReportView_Column.Table_Name);
+		if (m_Reportview_Column.getAD_ReportView_ID() <= PackOut.MAX_OFFICIAL_ID || m_Reportview_Column.getAD_Column_ID() <= PackOut.MAX_OFFICIAL_ID)
+			filler.add("AD_ReportView_Column_UU", new AttributesImpl());
+
+		filler.export(excludes);
+	}
+
+	@Override
+	public void packOut(PackOut packout, TransformerHandler packoutHandler, TransformerHandler docHandler, int recordId)
+			throws Exception {
+		throw new AdempiereException("X_AD_ReportView_Column doesn't have ID, use method with UUID");
+	}
+
+	@Override
+	public void packOut(PackOut packout, TransformerHandler packoutHandler,
+			TransformerHandler docHandler,
+			int recordId, String uuid) throws Exception {
+		X_AD_ReportView_Column po = new Query(packout.getCtx().ctx, X_AD_ReportView_Column.Table_Name, "X_AD_ReportView_Column_UU=?", getTrxName(packout.getCtx()))
+				.setParameters(uuid)
+				.first();
+		
+		if (po != null) {
+			Env.setContext(packout.getCtx().ctx, X_AD_ReportView_Column.COLUMNNAME_AD_ReportView_ID, po.getAD_ReportView_ID());
+			Env.setContext(packout.getCtx().ctx, X_AD_ReportView_Column.COLUMNNAME_AD_Column_ID, po.getAD_Column_ID());
+			this.create(packout.getCtx(), packoutHandler);
+			packout.getCtx().ctx.remove(X_AD_ReportView_Column.COLUMNNAME_AD_ReportView_ID);
+			packout.getCtx().ctx.remove(X_AD_ReportView_Column.COLUMNNAME_AD_Column_ID);
```

Comment:
should either pass the uuid to the create method through context or since you already have the po here, just pass it to the create method through context (you can use packout.getCtx().ctx.put()). it is unnecessary and inefficient to retrieve the po again at the create method.

Reply:
I think I can put the PO in the context in packOut method using
packout.getCtx().ctx.put("po", po);

But I don't see how I can retrieve it in the create method (as Env.getContext* methods can retrieve only int, String Timestamp but not Object)
What am I missing ?

Reply:
Instead of Env.getContext, you can use ctx.ctx.get(..)